test/e2e/locality_aware_lb_test.go

```diff
@@ -0,0 +1,249 @@
+//go:build integ
+// +build integ
+
+/*
+End-to-End Test for Kmesh Locality-Aware Load Balancing.
+
+This test verifies that Kmesh correctly routes traffic to the nearest available instance 
+based on locality (region, zone, and subzone). The test performs the following steps:
+1. Create a Kind cluster with three worker nodes.

```

Comment:
It's better to test in an existing E2E cluster ranther than creating a new Kind cluster.